DOCS-11165 ACM 2.0 (#223)
mulesoft/docs-api-community-manager: modules/ROOT/pages/update-acm.adoc

= Upgrades
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../assets/images/

Upgrades are applied automatically to Anypoint API Community Manager versions 2.0 and later.

Upgrades with new features are applied to sandbox environments four weeks before they are applied to production environments. Upgrades with only bug fixes are applied to sandbox environments two weeks before they are applied to production environments.

> To ensure the community administrator controls how their audience receives community changes, new functionality in upgrades is typically disabled until the community administrator enables it.

Reviewer: ensure that, always, per style ref:
```suggestion
To ensure that the community administrator controls how their audience receives community changes, new functionality in upgrades is typically disabled until the community administrator enables it.
```

Author: Thank you! I will try to remember this. Fixed.

.. Click *Save*.
. Verify that your permissions match those in the xref:create-community.adoc#configure-profiles-and-permissions[Configure Profiles and Permissions] documentation.

Return to the main list of upgrade instructions to continue.

[[post-20]]
== Postinstallation Steps When Upgrading from a Version Earlier Than 2.0

> If you are upgrading from a package version earlier than 2.0, use either the automatic or manual configuration methods to update the permissions of the member and guest profiles. Both profiles require field-level read access to the *AssetId* field in the *AnypointApiInstances* object.

Reviewer: ```suggestion
If you are upgrading from a package version earlier than 2.0, use either the automatic or manual configuration method to update the permissions of the member and guest profiles. Both profiles require field-level read access to the *AssetId* field in the *AnypointApiInstances* object.
```

Author: Fixed.

Automatic configuration:

. From the Anypoint API Community Manager control panel, open the Salesforce App Launcher and search for `API community manager diagnostics`.

> . In the *User Access* tab, in the section *Guest User*, click *Fix all*.
> . In the *User Access* tab, in the section *Member User*, click *Fix all*.

Reviewer: We could probably combine these:
```suggestion
. In the *User Access* tab, in the sections *Guest User* and *Member User*, click *Fix all*.
```

Author: Yes, that's better. Fixed.

> . Verify that your permissions match those in the xref:create-community.adoc#configure-profiles-and-permissions[Configure Profiles and Permissions] documentation.

Reviewer: Wait...you go through this whole task and then doublecheck it against some other doc?

Author: Yes. Customers have said the upgrade configurations are a pain point, and the ACM team is working on a comprehensive solution to automate all of this. The ACM team discussed whether to remove the comparison to the permissions on the community page, which increases the risk of errors, troubleshooting, or support cases, or keep the comparison, which makes the instructions more complex, and decided to keep it until the more comprehensive solution is released.

Manual configuration:

. Configure the member profile permissions:
.. Click *Setup* > *Profiles*.
.. Open the *Member* profile.
.. Click *Object settings* > *AnypointApiInstances*.
.. Ensure that the *AssetId* field has *Read* access.
.. Save.
. Configure the guest profile permissions:
.. Open the community builder.
.. Click *Settings* > *General*.
.. Open the *Guest* profile.
.. Click *Object settings* > *AnypointApiInstances*.
.. Ensure that the *AssetId* field has *Read* access.
.. Save.

> . Verify that your permissions match those in the xref:create-community.adoc#configure-profiles-and-permissions[Configure Profiles and Permissions] documentation.

Reviewer: ditto re. verification of instructions

Author: see below
